### ISSUE TITLE
KeyError: 'title' in MySQL engine and other engines using the key-value.html  template

### ISSUE THREAD

**Issue by return42** [searxng/searxng#3130]
Since #2508 a title is required --> this is a bug when an engine uses the [key-value.html](https://github.com/searxng/searxng/blob/master/searx/templates/simple/result_templates/key-value.html) where no title is needed.

------

### Discussed in https://github.com/searxng/searxng/discussions/3128

<div type='discussions-op-text'>

<sup>Originally posted by **kingmoe001** January 16, 2024</sup>
I am trying to use the MySQL engine on searxng to return results from my database. However, every time I run the webserver i get this error message:

```
* Serving Flask app 'webapp'
 * Debug mode: off
2024-01-16 16:35:39,637 ERROR:searx.webapp: 'title'
Traceback (most recent call last):
  File "/usr/local/searxng/searxng-src/searx/webapp.py", line 661, in search
    result_container = search.search()
  File "/usr/local/searxng/searxng-src/searx/search/__init__.py", line 212, in search
    self.result_container.close()
  File "/usr/local/searxng/searxng-src/searx/results.py", line 360, in close
    result['title'] = ' '.join(utils.html_to_text(result['title']).strip().split())
KeyError: 'title'
```

I believe the issue to be related to the MySQL engine because when I turn off the feature in searxng preferences, I can retrieve search results and this error message no longer pops up. 
![Screenshot (7)](https://github.com/searxng/searxng/assets/48866202/ea0d3dcd-16e0-4049-95d9-17b4b5886c48)

Here is my settings.yml file server and formats configuration. 
![Screenshot (8)](https://github.com/searxng/searxng/assets/48866202/a05aff99-569b-4db8-8cd7-6167d528966f)

and MySQL 
![Screenshot (9)](https://github.com/searxng/searxng/assets/48866202/2f82cc12-fc49-4ab8-b559-f54a635ab325)

Also, I have checked - the rate limiter plugin is disabled

Any suggestions?

</div>

**Comment by kingmoe001** [comment]
To clarify, does this imply that there is currently no resolution available for this issue? I apologize for any confusion, as I may not be familiar with developer terminology.

**Comment by return42** [comment]
We have to fix the issue .. as interim solution you can add a `title` column to your sql query.

**Comment by ahmad-alkadri** [comment]
Looks like a simple `try`-`except` could solve the problem, I think? Maybe we can simply add empty string in the title if there's no `title` column. So something like:

```python
try:
    result['title'] = ' '.join(utils.html_to_text(result['title']).strip().split())
except KeyError:
    result['title'] = ""
```

What do you think?

**Comment by SystemZ** [comment]
> Looks like a simple `try`-`except` could solve the problem, I think? Maybe we can simply add empty string in the title if there's no `title` column. So something like:
> 
> ```python
> try:
>     result['title'] = ' '.join(utils.html_to_text(result['title']).strip().split())
> except KeyError:
>     result['title'] = ""
> ```
> 
> What do you think?

I tried it when facing same problem with elasticsearch engine and it worked :)
I'll try to make PR for this until better solution would be implemented.

**Comment by return42** [comment]
> I'll try to make PR for this until better solution would be implemented.

@SystemZ @ahmad-alkadri I implemented PR #3187 to fix the issue / would give it a try .. thanks :+1:
